test/e2e/plugin_sync/plugin_sync_lifecycle_test.go

```diff
@@ -0,0 +1,370 @@
+
+	// Use case 1: create a KIND cluster, don't apply CRD and CRs, create context, make sure no plugins are installed
+	// a. create k8s context for the KIND cluster
+	// b. create context and validate current active context
+	// c. list plugins and make sure no plugins installed
+	// d. delete KIND cluster
+	Context("plugin install from group: install a plugin from a specific plugin group", func() {
+		var clusterInfo *framework.ClusterInfo
+		var contextName string
+		var err error
+		// Test case: a. create k8s context for the KIND cluster
+		It("create KIND cluster", func() {
+			// Create KIND cluster, which is used in test cases to create context's
+			clusterInfo, err = framework.CreateKindCluster(tf, "sync-e2e-"+framework.RandomNumber(4))
+			Expect(err).To(BeNil(), "should not get any error for KIND cluster creation")
+		})
+		// Test case: b. create context and validate current active context
+		It("create context with kubeconfig and context", func() {
+			By("create context with kubeconfig and context")
+			contextName = "sync-e2e-" + framework.RandomString(4)
+			err := tf.ContextCmd.CreateConextWithKubeconfig(contextName, clusterInfo.KubeConfigPath, clusterInfo.ClusterKubeContext)
+			Expect(err).To(BeNil(), "context should create without any error")
+			active, err := tf.ContextCmd.GetActiveContext(string(types.TargetK8s))
+			Expect(err).To(BeNil(), "there should be a active context")
+			Expect(active).To(Equal(contextName), "the active context should be recently added context")
+		})
+		// Test case: c. list plugins and make sure no plugins installed
+		It("list plugins and check number plugins should be same as installed in previous test", func() {
+			pluginsList, err := tf.PluginCmd.ListPluginsForGivenContext(contextName, true)
+			Expect(err).To(BeNil(), "should not get any error for plugin list")
+			Expect(len(pluginsList)).Should(Equal(0), "there should not be any context specific plugins")
+		})
+		// Test case: d. delete KIND cluster
+		It("delete the KIND cluster", func() {
+			err = tf.ContextCmd.DeleteContext(contextName)
+			Expect(err).To(BeNil(), "context should be deleted without error")
+			_, err := tf.KindCluster.DeleteCluster(clusterInfo.Name)
+			Expect(err).To(BeNil(), "kind cluster should be deleted without any error")
+		})
+	})
+
+	// Use case 2: Create kind cluster, apply CRD and CRs, create context, should install all plugins, uninstall the specific plugin, and perform plugin sync:
+	// Steps:
+	// a. create KIND cluster, apply CRD
+	// b. apply CRD (cluster resource definition) and CR's (cluster resource) for few plugins
+	// c. create context and make sure context has created
+	// d. list plugins and validate plugins info, make sure all plugins installed for which CR's has applied to KIND cluster
+	// e. uninstall one of the installed plugin, make sure plugin is uninstalled,
+	//		run plugin sync, make sure the uninstalled plugin has installed again.
+	// f. delete the KIND cluster
```

Comment:
```suggestion
	// f. delete the current context
	// g. list plugins for the context and it should not show any plugins
```

NOTE: We should add `testcase-g` mentioned above.